GETTING_STARTED.md

```diff
@@ -0,0 +1,98 @@
+# Introduction
+
+This doc is designed to help beginners get into Melee decompilation (and hopefully contribute). Also, we'll try to answer common questions here and talk about motivations for working on this project.
+
+# Goals of the project
+
+The goal of this repo is to **write C code that, after being compiled, matches the PowerPC assembly 100%**. There's a bit of nuance here though. Let’s first take a look at what we’re working with:
+
+![compilation_diagram](images/compilation_diagram.png)
+
+The details for what produces these different artifacts is out-of-scope. Here’s what some Melee C code looks like:
+
+```c
+void func_8007B8CC(Fighter* fp, HSD_GObj* otherObj) {
+    Fighter* otherFp;
+
+    otherFp = getFighter(otherObj);
+    ((u32 *)fp)[0x466] = (u32) otherObj;
+    fp->x119C_teamUnk = (u8) otherFp->x61B_team;
+    fp->x119D = (u8) otherFp->xC_playerID;
+}
+```
+
+and it’s corresponding PowerPC assembly:
+
+```asm
+.global func_8007B8CC
+func_8007B8CC:
+/* 8007B8CC 000784AC  80 A4 00 2C */	lwz r5, 0x2c(r4)
+/* 8007B8D0 000784B0  90 83 11 98 */	stw r4, 0x1198(r3)
+/* 8007B8D4 000784B4  88 05 06 1B */	lbz r0, 0x61b(r5)
+/* 8007B8D8 000784B8  98 03 11 9C */	stb r0, 0x119c(r3)
+/* 8007B8DC 000784BC  88 05 00 0C */	lbz r0, 0xc(r5)
+/* 8007B8E0 000784C0  98 03 11 9D */	stb r0, 0x119d(r3)
+/* 8007B8E4 000784C4  4E 80 00 20 */	blr
+```
+
+So the repo is filled with `.s` files and `.c` files, where the C code inside the `.c` files should have translated 100% to some contents in a `.s` file that used to be in the repo. Note that once we successfully decompile a function, we normally remove it from the assembly.
+
+# Do we have to write all this C code by hand?
+
+No! And even initially, the code doesn’t have to be super-readable as we’re trying to get 100% match rate first.
+
+So you can actually view the above example in an online tool that we use called decomp.me. Here’s a link to [that specific decompliation](https://decomp.me/scratch/MLlcK). If you want to try doing this yourself, follow these steps:
+
+1. Copy the assembly in the example
+2. Hit “New Scratch”
+3. Select Platform “GameCube / Wii”
+4. Set the preset to “Super Smash Bros. Melee”
+5. Copy in the contents of [this link](https://gist.github.com/stephenjayakar/73e9f7a1f77b9b2df281f0629b95290c) into the “Context” section. More on that later
```

Comment:
A static context will not be sufficient most of the time. Our tools for generating context are a bit rough around the edges at the moment, so we'll need to clean those up and add usage instructions to this document. You (@stephenjayakar) should take a look at `gen.sh` to see how we're doing that right now.

```suggestion
5. Copy in the contents of [this link](https://gist.github.com/stephenjayakar/73e9f7a1f77b9b2df281f0629b95290c) into the “Context” section. More on that later.
```

Reply:
uh... how exactly _do_ the commands in gen.sh work? I also didn't know at all about the tools/ directory -- some documentation around those would be helpful too.

Reply:
We'll have to open another PR to document those, it's something I've been meaning to do. This is fine for now.